cassiopeia/tools/tree.py

```diff
@@ -0,0 +1,227 @@
+from typing import List, Tuple
+
+import networkx as nx
+
+
+class Tree:
+    r"""
+    A phylogenetic tree for holding data from lineages and lineage tracing
+    experiments.
+
+    (Currently implemented as a light wrapper over networkx.DiGraph)
+
+    Args:
+        tree: The networkx.DiGraph from which to create the tree.
+    """
+
+    def __init__(self, tree: nx.DiGraph):
+        self.tree = tree
+
+    def root(self) -> int:
+        tree = self.tree
+        root = [n for n in tree if tree.in_degree(n) == 0][0]
+        return root
+
+    def leaves(self) -> List[int]:
+        tree = self.tree
+        leaves = [
+            n
+            for n in tree
+            if tree.out_degree(n) == 0 and tree.in_degree(n) == 1
+        ]
+        return leaves
```

Comment:
No need for the `and tree.in_degree(n) == 1` clause, in my opinion. In fact, if we do have any nodes that have more than one parent, we should be throwing an error because it's not a well-formed tree.

Reply:
Agreed!